test/test_datasets.py

```diff
@@ -102,29 +131,43 @@ def test_get_studies():
                           (None, None)
                           ])
 def test_subset_studies(all_datasets, fraction, num_studies):
-    seed = 42
     for dataset in all_datasets:
-
         if num_studies is None and fraction is None:
             with pytest.raises(ValueError):
-                dataset.subset_studies(fraction, num_studies, seed)
+                dataset.subset_studies(fraction, num_studies, seed=1)
             continue
 
+        samples = None
         if num_studies is not None:
-            subset_dataset = dataset.subset_studies(fraction, num_studies, seed)
+            subset_dataset = dataset.subset_studies(fraction, num_studies, seed=1)
+
+            samples = subset_dataset.get_samples()
             studies = subset_dataset.get_studies()
+
             assert len(studies) == num_studies
 
         else:
             all_samples = dataset.get_samples()
 
-            subset_dataset = dataset.subset_studies(fraction, num_studies, seed)
+            subset_dataset = dataset.subset_studies(fraction, num_studies, seed=1)
             samples = subset_dataset.get_samples()
 
             assert len(samples) > len(all_samples) * fraction
 
         dataset.reset_filters()
 
+        # There are only six studies, so tests will spuriously fail when selecting many studies
```

Comment:
Couldn't this still fail even when there's (for example) one study, if the one you select below happens to be the same as the one you selected before? Or am I missing something?

I guess if you're using the same seeds (1 and 2) every time and it doesn't fail the first time it won't fail in the future, so maybe this is fine. It seems like it could fail if you changed the seeds, though.

Reply:
It could still fail, but it doesn't with the given seeds. The second condition (the two subsets from the same seed should be the same) should always hold, so I guess it didn't strictly need to be in the if statement. On the other hand the first condition indicates success if it is ever true, because there has to be something nondeterministic going on to get different results. 

Ideally I'd like to just run the code fifty times with fifty different seeds, but there doesn't seem to be a built-in way to do that and I'm satisfied enough with it working now to not deal with the loop overhead